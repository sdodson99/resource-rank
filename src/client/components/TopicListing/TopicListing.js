import React from 'react';
import PropTypes from 'prop-types';
import styles from './TopicListing.module.css';
import TopicListingItem from '../TopicListingItem/TopicListingItem';
import Listing from '../Listing/Listing';

const TopicListing = ({ topics }) => {
  const topicListingItems = topics.map((t) => (
    <TopicListingItem
      key={t.id}
      name={t.name}
      slug={t.slug}
      verified={t.verified}
    />
  ));

  return (
    <div className={styles.TopicListing} data-testid="TopicListing">
      <Listing>{topicListingItems}</Listing>
    </div>
  );
};

TopicListing.propTypes = {
  topics: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      slug: PropTypes.string,
      verified: PropTypes.bool,
    })
  ),
};

TopicListing.defaultProps = {
  topics: [],
};

export default TopicListing;
