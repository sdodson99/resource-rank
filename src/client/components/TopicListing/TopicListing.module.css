.TopicListing {}

.TopicListingItem:last-child {
    @apply mx-5;
}