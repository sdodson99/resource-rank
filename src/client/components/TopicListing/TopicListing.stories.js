import React from 'react';
import TopicListing from './TopicListing';

export default {
  component: TopicListing,
  title: 'Components/TopicListing',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <TopicListing {...args} />;

export const Default = Template.bind({});
Default.args = {
  topics: [
    {
      name: 'Topic 1',
      slug: 'topic-1',
    },
    {
      name: 'Topic 2',
      slug: 'topic-2',
      verified: true,
    },
    {
      name: 'Topic 3',
      slug: 'topic-3',
    },
  ],
};

export const Single = Template.bind({});
Single.args = {
  topics: [
    {
      name: 'Topic 1',
      slug: 'topic-1',
    },
  ],
};

export const Empty = Template.bind({});
Empty.args = {};
