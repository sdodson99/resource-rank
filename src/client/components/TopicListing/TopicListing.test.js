import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TopicListing from './TopicListing';
import renderer from 'react-test-renderer';

describe('<TopicListing />', () => {
  test('it should mount', () => {
    render(<TopicListing />);

    const topicListing = screen.getByTestId('TopicListing');

    expect(topicListing).toBeInTheDocument();
  });

  describe('with multiple topics', () => {
    const topics = [
      {
        id: '123',
        name: 'Topic1',
      },
      {
        id: '456',
        name: 'Topic2',
      },
      {
        id: '789',
        name: 'Topic3',
      },
    ];

    it('should render correctly', () => {
      const tree = renderer.create(<TopicListing topics={topics} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('without topics', () => {
    it('should render correctly', () => {
      const tree = renderer.create(<TopicListing />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
