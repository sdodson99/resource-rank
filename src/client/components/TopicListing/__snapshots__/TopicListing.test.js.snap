// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicListing /> with multiple topics should render correctly 1`] = `
<div
  className="TopicListing"
  data-testid="TopicListing"
>
  <div
    className="border-b"
  >
    <div
      className="TopicListingItem"
      data-testid="TopicListingItem"
    >
      <a
        href="/topics/topic1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="ListingItem HoverListingItem"
          data-testid="ListingItem"
        >
          <div
            className="sm:flex sm:justify-between sm:text-left text-center"
          >
            <div>
              Topic1
            </div>
            <div
              className="mt-5 sm:mt-0"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="border-b"
  >
    <div
      className="TopicListingItem"
      data-testid="TopicListingItem"
    >
      <a
        href="/topics/topic2"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="ListingItem HoverListingItem"
          data-testid="ListingItem"
        >
          <div
            className="sm:flex sm:justify-between sm:text-left text-center"
          >
            <div>
              Topic2
            </div>
            <div
              className="mt-5 sm:mt-0"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className=""
  >
    <div
      className="TopicListingItem"
      data-testid="TopicListingItem"
    >
      <a
        href="/topics/topic2"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="ListingItem HoverListingItem"
          data-testid="ListingItem"
        >
          <div
            className="sm:flex sm:justify-between sm:text-left text-center"
          >
            <div>
              Topic3
            </div>
            <div
              className="mt-5 sm:mt-0"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<TopicListing /> without topics should render correctly 1`] = `
<div
  className="TopicListing"
  data-testid="TopicListing"
/>
`;
